Taking over: Adds dependencies, making a PR, and updating documentation to NDG (#1315)
pyne/pyne: docs/devsguide/new_dev_guide.rst

.. _devsguide_new_dev_guide:

*********************
New Developer's Guide
*********************

==================
Setting up Github:
==================

> Before you start using GitHub, you have to make Git available on your computer.

Reviewer: ```suggestion
Before you start using GitHub, you have to install Git on your computer.
```

> Even if it is on your device, it is a good idea to update to the latest version.

Reviewer: I don't think this sentence is needed

You can either install it as a package, via another installer or download the
source code and compile it yourself.

Download Git `here
<https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`__.

------------------------
Connecting GitHub to Git
------------------------

> To connect Git on your device to GitHub `here

Reviewer: ```suggestion
To connect Git on your device to GitHub, follow the instructions `here
```

Make sure that your device has all of the dependencies downloaded. You 
can check this by typing the name of the program into your command line 

> followed by "--version" (e.g., $ python --version). If you are starting as a

Reviewer 2: From the other PR, I had suggested to change $ python --version to have the double ` so that it renders as code.

Reviewer 3: That's really strange, @opotowsky I thought we fixed this in my PR by changing the line to 
```suggestion
followed by :code:`--version` (e.g., :code:`$ python --version`). If you are starting as a
```

Author: maybe I messed up :(

it is in now

compatible with the current version of PyNE, follow these steps to
create an environment for the conda install method.

Documentation on creating and managing an environment can be found
`here <https://docs.conda.io/projects/conda/en/latest/user-guide/
tasks/manage-environments.html>`__.

==========================
Signing up for list hosts:
==========================

Everyone faces challenges sometimes when writing code. Thankfully, you can always
contact the PyNE Developers at pyne-dev@groups.google.com. Another way to
communicate with PyNE developers is to join the `Google Group

> <https://groups.google.com/forum/#!forum/pyne-users>`__.

Reviewer: this is the user group not the dev group- is this intentional?

========================
Preparing to Contribute:
========================

The skills of forking and cloning are especially important to have mastered before
beginning any contribution to the repository.

Before forking this or any other repository, engage SSH keys. The process of

> creating SSH keys on your device is detailed

Reviewer: this is not at all a necessary step for developing on PyNE.

same tag. Pinoli is the Italian word for the Pine Nut, and this marker is the
first place New Developers should look to contribute.

---------------------
Making a Pull Request
---------------------
Before you make the pull request (PR), make sure that you include a change to the 
CHANGELOG.rst file under the appropriate heading. Your contribution can be as simple 
as a line of text, but it should incorporate the PR's number in parenthesis at the end. 
If there is a line already in the changelog that describes a similar action (e.g., 
adding a publication) to your own, you can add the number of your PR after the existing line.

The PR should contain a description of the changes, appropriate labels, projects,
and a reference to the issue that led to it, which you can do by inserting
‘#issue_number’ to the description (i.e., adding #161 to your PR description).
These elements will help other contributors discover and review your work. Adding a
reference to the issue the pull request will allow people to see the issue inspired
it alongside any conversation about the issue.

----------------------
Changing Documentation
----------------------

> To contribute, you can edit the text file in any program that allows you to edit

Reviewer 4: Presumably one needs an editor for any of the above contributions, as well, right?  Why mention it here?  Perhaps because we imagine documentation contributions as lower hanging fruit and expect folks to enter here?

Author: This existed before @nsryan2's update, 

I believe the emphasis was more on the no `MSWord` than on the editor.
